### DISCUSSION TITLE
Question about MBAR

### DISCUSSION THREAD

**Discussion by faranak1991** [openforcefield/discussions#33] [alchemiscale User Questions]
Hello, 

I'm interested in minimizing the objective function with a gradient optimization approach. While I comprehend how MBAR estimates and reweighs the property of interest, I'm currently puzzled about how MBAR calculates the derivative of the objective function (or the property of interest); this aspect isn't clear to me? 
Any help would be appreciated.

**Comment by mrshirts** [comment]
Probably a question for [pymbar](https://pymbar.readthedocs.io/en/master/), not here. The code there has moderately well-documented descriptions of the objective function and the calculation of the gradients and hessians - you can use any scipy minimizer (using gradients/hessians/etc) to solve MBAR.  Some of the papers of Levy, Tan, et al that call the same equation UWHAM (unbinned weighted histogram analysis method) go through the minimization in more detail.

**Comment by faranak1991** [comment]
Dear Dr. Shirts, I really appreciate your reply.
The challenge I'm facing is that I intend to improve the force field parameters for my system using a gradient-based optimizer. However, this requires calculating the gradient, which happens to be the costly component of the procedure. This is why I'm considering utilizing MBAR to compute the gradient that can be fed to the optimizer. 
Is it possible to make small perturb to the parameters during each iteration, subsequently utilizing MBAR to estimate the properties for these altered parameters, and then computing the gradient with respect to each parameter?

**Comment by mrshirts** [comment]
Evaluator does have the capability to estimate finite differences in free energies and expectations using MBAR, by evaluating the energy with reweighting at around the current point.  @SimonBoothroyd can probably clarify how to access that functionality.

**Comment by SimonBoothroyd** [comment]
@faranak1991 does https://docs.openforcefield.org/projects/evaluator/en/stable/properties/gradients.html help?
